cmd/cstor-pool-mgmt/controller/pool-controller/handler.go

```diff
@@ -161,7 +169,44 @@ func (c *CStorPoolController) cStorPoolEventHandler(operation common.QueueOperat
 	return string(apis.CStorPoolStatusInvalid), nil
 }
 
-// cStorPoolAddEvent does import of pool, and, if it fails, attemps to create pool based on CSP CR state
+// cStorPoolCreate does create of pool, and, if it fails, returns CreateFailed
+func (c *CStorPoolController) cStorPoolCreate(cStorPoolGot *apis.CStorPool) (string, error) {
+	if !IsCStorPoolCreateStatuses(cStorPoolGot) {
+		c.recorder.Event(cStorPoolGot, corev1.EventTypeWarning, string(common.FailureCreate), string(common.MessageImproperPoolStatus))
+		return string(cStorPoolGot.Status.Phase), fmt.Errorf("invalid status %s for create pool %s", cStorPoolGot.Status.Phase, cStorPoolGot.Name)
+	}
+
+	devIDList, err := c.getDeviceIDs(cStorPoolGot)
+	if err != nil {
+		return string(apis.CStorPoolStatusCreateFailed), errors.Wrapf(err, "failed to get device id of disks for csp %s", cStorPoolGot.Name)
+	}
+
+	// ValidatePool is to check if pool attributes are correct.
+	err = pool.ValidatePool(cStorPoolGot, devIDList)
+	if err != nil {
+		c.recorder.Event(cStorPoolGot, corev1.EventTypeWarning, string(common.FailureValidate), string(common.MessageResourceFailValidate))
+		return string(apis.CStorPoolStatusCreateFailed), err
+	}
+
+	// IsInitStatus is to check if initial status of cstorpool object is `init`.
+	if len(common.InitialImportedPoolVol) != 0 {
+		glog.Errorf("failed to handle add event: invalid status %v for pool %v with existing volumes", string(cStorPoolGot.Status.Phase), string(cStorPoolGot.GetUID()))
+		c.recorder.Event(cStorPoolGot, corev1.EventTypeWarning, string(common.FailureValidate), string(common.MessageImproperPoolStatus))
+		return string(apis.CStorPoolStatusCreateFailed), err
+	}
+	// CreatePool is to create cstor pool.
+	err = pool.CreatePool(cStorPoolGot, devIDList)
+	if err != nil {
+		glog.Errorf("Pool creation failure: %v", string(cStorPoolGot.GetUID()))
+		c.recorder.Event(cStorPoolGot, corev1.EventTypeWarning, string(common.FailureCreate), string(common.MessageResourceFailCreate))
+		return string(apis.CStorPoolStatusCreateFailed), err
+	}
+	glog.Infof("Pool creation successful: %v", string(cStorPoolGot.GetUID()))
+	c.recorder.Event(cStorPoolGot, corev1.EventTypeNormal, string(common.SuccessCreated), string(common.MessageResourceCreated))
+	return string(apis.CStorPoolStatusOnline), nil
```

Comment:
So we need to log and event simultaneously.

Can we do something to avoid repetition of this logic.

Reply:
yes.. we need to avoid this.. but, being old code that gets deprecated, we can leave it here